### PR TITLE
Feat: Add Audience + Scopes Param to PKCE and DeviceCode Auth Flows

### PR DESCRIPTION
**Pull request by PudgyPigeon**
# TL;DR
This PR adds the `audience` parameter into the PKCE/AuthorizationClient and DeviceCodeAuthenticator auth flows. This param is derived from `auth_helper.RemoteClientConfigStore.get_client_config()`

Also, the `scopes` configuration - derived from `config.yaml` by way of  `configuration.PlatformConfig` is now exposed to the aforementioned auth flows. 

Changes are also made to the way the scopes parameter is parsed in the `token_client` module to bring it in line with the AuthorizationClient class flow.

This now makes it so that Auth0 can use the PKCE and DeviceCode Auth flows without errors.

Caching + refresh tokens is not included in this PR but a new one can be made shortly.

## Type
 - [x] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [x ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This PR adds the `audience` parameter into the PKCE/AuthorizationClient and DeviceCodeAuthenticator auth flows. This param is derived from `auth_helper.RemoteClientConfigStore.get_client_config()`

Also, the `scopes` configuration - derived from `config.yaml` by way of  `configuration.PlatformConfig` is now exposed to the aforementioned auth flows. 

Changes are also made to the way the scopes parameter is parsed in the `token_client` module to bring it in line with the AuthorizationClient class flow.

This now makes it so that Auth0 can use the PKCE and DeviceCode Auth flows without errors.

Caching + refresh tokens is not included in this PR but a new one can be made shortly.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by PudgyPigeon** [comment]
@wild-endeavor @kumare3 

I made the requested changes to the previous PR to separate caching/refresh-tokens and the scopes+audience for Auth0/DeviceCode flows.

Hopefully this is up to standard

**Comment by PudgyPigeon** [comment]
@wild-endeavor 

Fixed the linting error thrown so should be fine. It seems however that two code coverage workflows are failing. How would you like to proceed on that front?

File: flytekit/clients/auth/auth_client.py [hunk 1]
```diff
@@ -238,6 +241,7 @@ def __init__(
         self._request_auth_code_params = {
             "client_id": client_id,  # This must match the Client ID of the OAuth application.
             "response_type": "code",  # Indicates the authorization code grant
+            "audience": self._audience,
```
Comment (wild-endeavor) [inline flytekit/clients/auth/auth_client.py:244]: could we add this only optionally if it's not None?

**Comment by PudgyPigeon** [comment]
@wild-endeavor PR should be ready for the Github Actions jobs

**Review by wild-endeavor — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]
